src/PhpGenerator/ClassLike.php

```diff
@@ -39,15 +39,27 @@ abstract class ClassLike
 
 	public static function from(string|object $class, bool $withBodies = false): static
 	{
-		return (new Factory)
+		$class = (new Factory)
 			->fromClassReflection(new \ReflectionClass($class), $withBodies);
+
+        if (!$class instanceof static) {
+            throw new Nette\InvalidArgumentException("Object '$class' is not an instance of " . static::class);
```

Comment:
The error message should be different so that the user understands what he is doing wrong. And please create a test case for it.

Reply:
Sure. I'm thinking of a message that provides better guidance:

> "Trait1" is not a "ClassType". Use TraitType::from() or ClassLike::from() instead.

Reply:
Test added and exception message improved. Is that all right?